velox/expression/VectorFunction.h

```diff
@@ -178,6 +178,7 @@ std::shared_ptr<const Type> resolveVectorFunction(
 /// constant input; non-constant inputs should be represented as nullptr;
 /// constant inputs must be instances of ConstantVector.
 std::shared_ptr<VectorFunction> getVectorFunction(
+    const core::QueryConfig& config,
```

Comment:
Move config to be the last argument for consistency.

Reply:
To consist with sinple funtion, I put it to be first argument.
```
      auto func = simpleFunctionEntry->createFunction()->createVectorFunction(
          config, getConstantInputs(compiledInputs));
```

```
    if (auto specialForm = getSpecialForm(
            config,
            call->name(),
            resultType,
            std::move(compiledInputs),
            trackCpuUsage)) {
      result = specialForm;
    } else if (
        auto func = getVectorFunction(
            config,
            call->name(),
            inputTypes,
            getConstantInputs(compiledInputs))) {
      result = std::make_shared<Expr>(
          resultType,
          std::move(compiledInputs),
          func,
          call->name(),
          trackCpuUsage);
    } else if (
        auto simpleFunctionEntry =
            simpleFunctions().resolveFunction(call->name(), inputTypes)) {
      VELOX_USER_CHECK(
          resultType->equivalent(*simpleFunctionEntry->type().get()),
          "Found incompatible return types for '{}' ({} vs. {}) "
          "for input types ({}).",
          call->name(),
          simpleFunctionEntry->type(),
          resultType,
          folly::join(", ", inputTypes));
      auto func = simpleFunctionEntry->createFunction()->createVectorFunction(
          config, getConstantInputs(compiledInputs));
```

Do I need to move it？

Reply:
I see. Let's look at all the APIs to create functions and make sure they are consistent. Would you compile such a list so we can get a better picture?

Reply:
I find 3:
```
  void createWindowFunctions(
      const std::shared_ptr<const core::WindowNode>& windowNode,
      const RowTypePtr& inputType,
      const core::QueryConfig& config);

class SimpleFunctionAdapterFactory {
 public:
  virtual std::unique_ptr<VectorFunction> createVectorFunction(
      const core::QueryConfig& config,
      const std::vector<VectorPtr>& constantInputs) const = 0;
  virtual ~SimpleFunctionAdapterFactory() = default;
};

std::shared_ptr<VectorFunction> getVectorFunction(
    const core::QueryConfig& config,
    const std::string& name,
    const std::vector<TypePtr>& inputTypes,
    const std::vector<VectorPtr>& constantInputs);
```

Do I need to move WindowFunction `config` to first argument? Or move all the config to last argument?

Reply:
If `Aggregate::create` need to aligh.
```
std::unique_ptr<Aggregate> Aggregate::create(
    const std::string& name,
    core::AggregationNode::Step step,
    const std::vector<TypePtr>& argTypes,
    const TypePtr& resultType,
    const core::QueryConfig& config) {
```

Reply:
Thanks. Let's move 'config' to the end of the argument list in all these APIs.

Reply:
Moved